### PR TITLE
[WIP] akka persistence query for SQL journals

### PR DESCRIPTION
**Pull request by Horusiath**

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@cconstantin Chris, would you be able to take a look at this?

**Comment by cconstantin** [comment]
Sure. It was WIP last time I looked.
-- 
Chris Constantin
Aligned Software Solutions Inc. 

T: 604 259 7676 | C: 778 838 1550 | aligned.io




On Tue, Jun 7, 2016 at 2:42 PM -0700, "Aaron Stannard" <notifications@github.com> wrote:












@cconstantin Chris, would you be able to take a look at this?

**Comment by cconstantin** [comment]
@Horusiath is this ready for review, if not for merge?

**Comment by Horusiath** [comment]
@cconstantin ready for review, however I don't know if we are going introduce this as part of v1.1.

**Comment by alexvaluyskiy** [comment]
@Horusiath despite this feature is very anticipated, I think we should not ship it in this release. We have already have a lot of features in this release.
And if we merge it, we should update all providers to support Persistent Query.

File: src/contrib/persistence/Akka.Persistence.Query.Sql.Tests/EventsByPersistenceIdSpec.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        protected EventsByPersistenceIdSpec(Config config, ITestOutputHelper output) : base(config, output: output)
+        {
+            _materializer = Sys.Materializer();
+            _queries = Sys.ReadJournalFor<SqlReadJournal>(SqlReadJournal.Identifier);
+        }
+
+        [Fact]
+        public void Sql_query_EventsByPersistenceId_should_implement_standard_EventsByTagQuery()
+        {
+            (_queries is IEventsByTagQuery).Should().BeTrue();
```
Comment (cconstantin) [inline src/contrib/persistence/Akka.Persistence.Query.Sql.Tests/EventsByPersistenceIdSpec.cs]: Shouldn't this check for `IEventsByPersistenceId`?

File: src/contrib/persistence/Akka.Persistence.Query.Sql/reference.conf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿akka.persistence.query.journal.sql {
+  # Implementation class of the SQL ReadJournalProvider
+  class = "Akka.Persistence.Query.Sql.SqlReadJournalProvider, Akka.Persistence.Query.Sql"
+  
+  # Absolute path to the write journal plugin configuration entry that this 
+  # query journal will connect to. That must be a LeveldbJournal or SharedLeveldbJournal.
```
Comment (cconstantin) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/reference.conf]: We don't have a LeveldbJournal, do we?

**Comment by Horusiath** [comment]
Is there something wrong with multinode tests?

**Comment by Aaronontheweb** [comment]
MNTR failure due to https://github.com/akkadotnet/akka.net/pull/2118

**Comment by Aaronontheweb** [comment]
If you rebase this again @Horusiath then it should all work since #2118 was merged in

**Pull request merged**